### PR TITLE
[TEST] Add tests for tandem duplications

### PR DESCRIPTION
**Pull request by Irallia**
Resolves a first part of #143
Some code refactoring and new test cases.

### CODE REVIEW

**Review by SGSSGene — APPROVED** [review]
:+1:

File: include/modules/sv_detection_methods/analyze_split_read_method.hpp [hunk 1]
```diff
@@ -36,6 +36,32 @@ void split_string(std::string const & str, Container & cont, char const delim =
  */
 void retrieve_aligned_segments(std::string const & sa_string, std::vector<AlignedSegment> & aligned_segments);
 
+/*! \brief Calculates the end respectively start position of two consecutive mates (current and next
+ *         [AlignedSegment](\ref AlignedSegment)) depending on their orientation and corrects the position of the second
+ *         mate, if they are overlapping.
+ *
+ * \param[in] current          - current [AlignedSegment](\ref AlignedSegment)
+ * \param[in] next             - consecutive next [AlignedSegment](\ref AlignedSegment)
+ * \param[in] distance_on_read - distance between the consecutive aligned_segments on the read
```
Comment (joergi-w) [inline include/modules/sv_detection_methods/analyze_split_read_method.hpp:45]: Use `AlignedSegments` as above or
```suggestion
 * \param[in] distance_on_read - distance between the consecutive aligned segments on the read
```

File: include/modules/sv_detection_methods/analyze_split_read_method.hpp [hunk 2]
```diff
@@ -36,6 +36,32 @@ void split_string(std::string const & str, Container & cont, char const delim =
...
+                       | |
+               mate1_pos mate2_pos
+
+    overlapping Case: --current-->         distance_on_read < 0
+                                --next-->
+                                 ||
+                         mate1_pos mate2_pos
+    \endverbatim
+ */
+std::tuple<int32_t, int32_t> get_mate_positions(AlignedSegment current, AlignedSegment next, int32_t distance_on_read);
```
Comment (joergi-w) [inline include/modules/sv_detection_methods/analyze_split_read_method.hpp:63]:
```suggestion
std::pair<int32_t, int32_t> get_mate_positions(AlignedSegment current, AlignedSegment next, int32_t distance_on_read);
```

File: src/modules/sv_detection_methods/analyze_split_read_method.cpp [hunk 3]
```diff
@@ -53,14 +53,42 @@ void retrieve_aligned_segments(std::string const & sa_string, std::vector<Aligne
     }
 }
 
+std::tuple<int32_t, int32_t> get_mate_positions(AlignedSegment current, AlignedSegment next, int32_t distance_on_read)
```
Comment (joergi-w) [inline src/modules/sv_detection_methods/analyze_split_read_method.cpp:56]:
```suggestion
std::pair<int32_t, int32_t> get_mate_positions(AlignedSegment current, AlignedSegment next, int32_t distance_on_read)
```

File: test/api/detection_test.cpp [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #include <gtest/gtest.h>
 
 #include <seqan3/alphabet/cigar/cigar.hpp>
+// #include <seqan3/core/debug_stream.hpp>  // include for debugging
```
Comment (joergi-w) [inline test/api/detection_test.cpp:4]: delete this?

**Review by joergi-w — CHANGES_REQUESTED** [review]

File: include/modules/sv_detection_methods/analyze_split_read_method.hpp (hunk 2 above)
Comment (joergi-w) [reply include/modules/sv_detection_methods/analyze_split_read_method.hpp:63]: Just as a note to think about: Actually, this function doesn't need to be declared in the header file, if it is only accessed locally from the same cpp file. Then, of course, it cannot be included/used from other files or tests.

File: test/api/detection_test.cpp (hunk 4 above)
Comment (Irallia) [reply test/api/detection_test.cpp:4]: I left this in the code, as I left some commented out debug stream lines in code. I don't know, if removing this line would be nicer.

**Review by joergi-w — APPROVED** [review]
Thank you :)

**Pull request merged**